rs/replicated_state/src/canister_state/queues/input_schedule.rs

```diff
@@ -0,0 +1,325 @@
+#![allow(dead_code)]
+
+use super::queue::CanisterQueue;
+use super::CanisterQueues;
+use crate::{InputQueueType, InputSource};
+use ic_protobuf::proxy::ProxyDecodeError;
+use ic_protobuf::state::queues::v1::canister_queues::NextInputQueue;
+use ic_protobuf::types::v1 as pb_types;
+use ic_types::CanisterId;
+use ic_validate_eq::ValidateEq;
+use ic_validate_eq_derive::ValidateEq;
+use std::collections::{BTreeSet, VecDeque};
+use std::convert::TryFrom;
+
+#[cfg(test)]
+mod tests;
+
+/// Round-robin schedule for canister inputs, across ingress, local senders and
+/// remote senders; as well as within local senders and remote senders.
+///
+/// We define three buckets of queues ("sources"): messages from canisters on
+/// the same subnet ("local senders"); ingress; and messages from canisters on
+/// other subnets ("remote senders"). Each time an input is popped or skipped,
+/// we rotate round-robin between the three sources. We also rotate round-robin
+/// over the canisters in the local sender and remote serder buckets when we pop
+/// or skip messages from those sources.
+#[derive(Clone, Debug, Default, PartialEq, Eq, ValidateEq)]
+pub(super) struct InputSchedule {
+    /// The input source (local senders, ingress or remote senders) at the front
+    /// of the schedule.
+    ///
+    /// Used to implement round-robin rotation over input sources.
+    #[validate_eq(Ignore)]
```

Comment:
Why do we have this as `Ignore`?

Reply:
No idea, I wanted to ask the same question when I first saw it.

@pakhomov-dfinity?